XenAdmin/Commands/RebootVMCommand.cs

```diff
@@ -198,17 +198,20 @@ protected override string GetCantExecuteReasonCore(SelectedItem item)
                 case vm_power_state.unknown:
                     return base.GetCantExecuteReasonCore(item);
             }
-            
-            if (vm.virtualisation_status == VM.VirtualisationStatus.PV_DRIVERS_NOT_INSTALLED || vm.virtualisation_status == VM.VirtualisationStatus.UNKNOWN)
+
+            if (vm.virtualisation_status.HasFlag(VM.VirtualisationStatus.PV_DRIVERS_NOT_INSTALLED) || vm.virtualisation_status.HasFlag(VM.VirtualisationStatus.UNKNOWN))

```

Comment:
This is a bug, which occurs a few times. PV_DRIVERS_NOT_INSTALLED == 0, and HasFlag(0) will always return true.

If you want to test for zero, you should use == 0 or .Equals(0). But before doing that, you should think about whether you really want to test for both I/O and Management to be absent, as 0 would do, or whether you just want to test for one or other of them missing.